app/assets/javascripts/admin/addon/components/theme-upload-add.js

```diff
@@ -0,0 +1,126 @@
+import Component from "@glimmer/component";
+import { action } from "@ember/object";
+import { tracked } from "@glimmer/tracking";
+import I18n from "I18n";
+import { ajax } from "discourse/lib/ajax";
+import { isEmpty } from "@ember/utils";
+
+const THEME_FIELD_VARIABLE_TYPE_IDS = [2, 3, 4];
+const SCSS_VARIABLE_NAMES = [
+  // common/foundation/colors.scss
+  "primary",
+  "secondary",
+  "tertiary",
+  "quaternary",
+  "header_background",
+  "header_primary",
+  "highlight",
+  "danger",
+  "success",
+  "love",
+  // common/foundation/math.scss
+  "E",
+  "PI",
+  "LN2",
+  "SQRT2",
+  // common/foundation/variables.scss
+  "small-width",
+  "medium-width",
+  "large-width",
+  "google",
+  "instagram",
+  "facebook",
+  "cas",
+  "twitter",
+  "github",
+  "base-font-size",
+  "base-line-height",
+  "base-font-family",
+  "primary-low",
+  "primary-medium",
+  "secondary-low",
+  "secondary-medium",
+  "tertiary-low",
+  "quaternary-low",
+  "highlight-low",
+  "highlight-medium",
+  "danger-low",
+  "danger-medium",
+  "success-low",
+  "love-low",
+];
+
+export default class ThemeUploadAdd extends Component {
+  @tracked name;
+  @tracked fileSelected = false;
+  @tracked flash;
+
+  get enabled() {
+    return this.nameValid && this.fileSelected;
+  }
+
+  get disabled() {
+    return !this.enabled;
+  }
```

Comment:
or… continuing the same thought - since `enabled` is used only in `disabled` I'd move all the logic to `disabled` 😃